### PR TITLE
Add test coverage for BZ:1982698

### PR DESCRIPTION
**Pull request by Gauravtalreja1**
### Problem Statement
Missing coverage for BZ:1982698, where we verified ansible playbook job template with "hosts: localhost:

### Solution
Add  test coverage for BZ:1982698, to verifiy ansible-playbook job template with "hosts: localhost"

### Dependent PRs:
https://github.com/SatelliteQE/nailgun/pull/1139

### CODE REVIEW

**Comment by Gauravtalreja1** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_ansible.py::TestAnsibleREX::test_positive_ansible_localhost_job_on_host
nailgun: 1139

**Comment by Gauravtalreja1** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_ansible.py::TestAnsibleREX::test_positive_ansible_localhost_job_on_host
nailgun: 1139

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 6871
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/api/test_ansible.py::TestAnsibleREX::test_positive_ansible_localhost_job_on_host --external-logging
Test Result : ================= 3 passed, 205 warnings in 1675.04s (0:27:55) =================
```

File: tests/foreman/api/test_ansible.py [hunk 1]
```diff
@@ -470,3 +470,63 @@ def test_positive_ansible_job_on_multiple_host(
...
+        :steps:
+            1. Register a content host with satellite
+            2. Run the Ansible playbook with "hosts: localhost" on registered host
+            3. Check if the job is executed and verify job output
+
+        :expectedresults:
+            1. Ansible playbook with "hosts: localhost" job execution must be successful.
+
+        :BZ: 1982698
+        """
```
Comment (shweta83) [inline tests/foreman/api/test_ansible.py:492]: :customerscenario: true

File: tests/foreman/api/test_ansible.py [hunk 2]
```diff
@@ -470,3 +470,63 @@ def test_positive_ansible_job_on_multiple_host(
...
+                'targeting_type': 'static_query',
+                'search_query': f'name = {rhel_contenthost.hostname}',
+                'inputs': {'playbook': playbook},
+            },
+        )
+        target_sat.wait_for_tasks(
+            f'resource_type = JobInvocation and resource_id = {job["id"]}', poll_timeout=1000
+        )
+        result = target_sat.api.JobInvocation(id=job['id']).read()
+        assert result.pending == 0
```
Comment (shweta83) [inline tests/foreman/api/test_ansible.py:526]: wait_for_tasks checks whether the task is completed or not. We can skip this assertion

File: tests/foreman/api/test_ansible.py [hunk 3]
```diff
@@ -470,3 +470,63 @@ def test_positive_ansible_job_on_multiple_host(
...
+        assert result.succeeded == 1
+        assert result.status_label == 'succeeded'
```
Comment (shweta83) [inline tests/foreman/api/test_ansible.py:528]: Any one of these assertion would work.

File: tests/foreman/api/test_ansible.py (hunk 2 above)
Comment (Gauravtalreja1) [reply tests/foreman/api/test_ansible.py:526]: We need to verify if job is succeccedd, and not pending after its completed as per BZ, so I'd keep this assertions

File: tests/foreman/api/test_ansible.py (hunk 3 above)
Comment (Gauravtalreja1) [reply tests/foreman/api/test_ansible.py:528]: I think better to check more assertions here

**Review by shweta83 — APPROVED** [review]
Ack

**Review by shubhamsg199 — APPROVED** [review]
ACK

**Pull request merged**